From 25ccea4c3b2d2e02a8fe376712672116ab37ca49 Mon Sep 17 00:00:00 2001
From: Pardha Kesapragada <pkesapragada@habana.ai>
Date: Wed, 23 Dec 2020 19:33:35 +0200
Subject: [PATCH] [SW-20154] goya2_arc: change dma size shift

The define for DMA_SIZE_SHIFT has been renamed in
Gaudi2, to MEM_SIZE_SHIFT instead of FLD_SIZE_SHIFT.
Hence, to accommodate such a change, get the
value of the shift from a platform specific function.

Change-Id: Ie57dfde653f3440ed4b8d94a6dce237059af45f1
Signed-off-by: Pardha Kesapragada <pkesapragada@habana.ai>
---

diff --git a/common/common.h b/common/common.h
index 674f4ba..812b9b1 100644
--- a/common/common.h
+++ b/common/common.h
@@ -9,5 +9,6 @@
 uint32_t get_mem_cfg_mask(void);
 uint8_t get_sris_mode_from_strap(void);
 void enable_flr_handling_fsm(void);
+uint32_t get_fld_size_shift(void);
 
 #endif /* _COMMON_H_ */
diff --git a/common/dma.c b/common/dma.c
index 63532d5..a49eb94 100644
--- a/common/dma.c
+++ b/common/dma.c
@@ -7,6 +7,7 @@
 
 #include "arcboot.h"
 #include "dma.h"
+#include <common.h>
 #include "spi_flash.h"
 
 static void dma_copy_image_internal(uint16_t image_size, uint32_t flash_offset,
@@ -17,6 +18,7 @@
 	uint32_t spi_dma_dst_addr_h = DMA_ADDRESS_HIGH(dest_offset);
 	uint16_t flash_offset_1k = flash_offset / DMA_SRC_BLOCK_SIZE;
 	uint16_t image_size_4b = (image_size / DMA_DST_BLOCK_SIZE);
+	uint32_t fld_size_shift = get_fld_size_shift();
 
 	regmap_writel(spi_dma_dst_addr_l,
 		      mmPSOC_GLOBAL_CONF_SPI_DMA_DST_ADDR_L);
@@ -30,8 +32,7 @@
 	reg_spi_dma_ctrl_s |= fw_sram_dst;
 
 	/* # of 4bytes to DMA copy */
-	reg_spi_dma_ctrl_s |= (uint32_t)image_size_4b <<
-			      PSOC_GLOBAL_CONF_SPI_DMA_CTRL_FLD_SIZE_SHIFT;
+	reg_spi_dma_ctrl_s |= (uint32_t)image_size_4b << fld_size_shift;
 
 	/* spi offset in 1k chunks */
 	reg_spi_dma_ctrl_s |= (uint32_t)flash_offset_1k <<
diff --git a/goya2_arc/common.c b/goya2_arc/common.c
index 48c2967..7f230ad 100644
--- a/goya2_arc/common.c
+++ b/goya2_arc/common.c
@@ -21,6 +21,11 @@
 	return PSOC_GLOBAL_CONF_MEM_CPY_CTRL2_FLD_SIZE_MASK;
 }
 
+uint32_t get_fld_size_shift(void)
+{
+	return PSOC_GLOBAL_CONF_SPI_DMA_CTRL_FLD_SIZE_SHIFT;
+}
+
 uint8_t get_sris_mode_from_strap(void)
 {
 	uint32_t boot_strap;
